dts/bindings/gpio/zephyr,gpio-keys.yaml

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2022 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: Zephyr GPIO KEYS parent node
+
+compatible: "zephyr,gpio-keys"
```

Comment:
Looking at how the binding is structured, I'm a bit concerned about the fact that this ends up allocating a `struct device` for every button. How about instead of this, we use a single device for the feature with a single debounce delay configuration, and thene specifiy the gpios as child nodes? Maybe you could build on top of the existing "gpio-keys" binding, then we'd have a single "debounced gpio" device, similarly to how it's been done for the "led-gpio" one: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/led/led_gpio.c

Reply:
I looked at using child nodes for this, but had concerns with looping to identify which node was triggering an interrupt, or getting the pin value. Since I opted to omit a key code in the dts, I think it makes it a little more difficult to identify which node is being affected by the API.  

The led_gpio example you provided takes an additional param in the API to identify the correct led node.  I vaguely recall the ordering in the device tree wasn't deterministic in code, is that correct?  I'll admit I'm not an expert in zephyr driver development, so if there's a best practice for this, I'm all for it :-)

Reply:
The ordering should be stable, otherwise the whole LED thing would not work (the API takes the LED as an index), it outputs something like:

`#define DT_N_S_buttons_FOREACH_CHILD(fn) fn(DT_N_S_buttons_S_button_0) fn(DT_N_S_buttons_S_button_1) fn(DT_N_S_buttons_S_button_2) fn(DT_N_S_buttons_S_button_3)`

How to address them is an interesting point, but you could either use the GPIO macros or make some equivalent one, and then the application can define a specific node for describing the high level scope of the gpio, which it'll likely do anyway, let's say

```
volume-buttons {
        compatible = "cros,volume-buttons";
        volume-up-gpios = <&debounced_gpios 1>
        volume-up-gpios = <&debounced_gpios 2>
}
```

The resource usage is something like 28 bytes per `struct device`, negligible for a couple buttons but imagine an application with a full panel to debounce, it would be significant.

Reply:
While using a DT chosen node to represent the key-mapping is a clever way of doing things, it poses an additional scalability problem given that this mapping would mean that there can be only 1 device capable of reporting a given input event.

2023 probably won't be The Year of the Zephyr Desktop (TM) or anything like that, but we should probably just adopt the `linux,code` property, or something like it.

Of course, there is nothing preventing you from creating a specific driver that reports CrOS-specific input events, but if we adopted the keycode approach, then at least we are leaving room for multiple input devices to be able to report the same events.

https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/input/gpio-keys.yaml